src/agent/src/storage/dm_verity.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::path::Path;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Context, Result};
+use image_rs::verity::umount_image_block_with_integrity;
+use kata_sys_util::mount::create_mount_destination;
+use kata_types::mount::KataVirtualVolume;
+use kata_types::mount::{DmVerityInfo, StorageDevice};
+use kata_types::volume::{
+    KATA_VOLUME_DMVERITY_OPTION_SOURCE_TYPE, KATA_VOLUME_DMVERITY_OPTION_VERITY_INFO,
+    KATA_VOLUME_DMVERITY_SOURCE_TYPE_PMEM, KATA_VOLUME_DMVERITY_SOURCE_TYPE_SCSI,
+    KATA_VOLUME_DMVERITY_SOURCE_TYPE_VIRTIO_CCW, KATA_VOLUME_DMVERITY_SOURCE_TYPE_VIRTIO_MMIO,
+    KATA_VOLUME_DMVERITY_SOURCE_TYPE_VIRTIO_PCI,
+};
+use protocols::agent::Storage;
+use slog::Logger;
+use tracing::instrument;
+
+use crate::storage::block_handler::{
+    PmemHandler, ScsiHandler, VirtioBlkCcwHandler, VirtioBlkMmioHandler, VirtioBlkPciHandler,
+};
+use crate::storage::{StorageContext, StorageHandler};
+
+#[derive(Debug)]
+pub struct DmVerityHandler {}
+
+impl DmVerityHandler {
+    fn get_dm_verity_info(storage: &Storage) -> Result<DmVerityInfo> {
+        for option in storage.driver_options.iter() {
+            if let Some((key, value)) = option.split_once('=') {
+                if key == KATA_VOLUME_DMVERITY_OPTION_VERITY_INFO {
+                    let virt_volume: KataVirtualVolume = KataVirtualVolume::from_base64(value)?;
```

Comment:
When passing the verity info through "driver_options", there's no need fro base64 encoding:)
We use base64 encoding for mount option because it will be parsed by containerd etc.

Reply:
Why passing `KataVirtualVolume` instead of `DmVerityInfo` here?
The whole `KataVirtualVolume` is converted to a `dm-verity` storage device, so the driver option should be `dm-verity info` instead of `KataVirtualVolume`.

Reply:
Done.